aries_vcx/src/handlers/issuance/mediated_issuer.rs

```diff
@@ -0,0 +1,112 @@
+use crate::errors::error::VcxResult;
+use crate::handlers::connection::mediated_connection::MediatedConnection;
+use crate::handlers::issuance::issuer::Issuer;
+use crate::handlers::util::{matches_opt_thread_id, matches_thread_id};
+use crate::protocols::issuance::issuer::state_machine::{IssuerFullState, IssuerSM, IssuerState};
+use agency_client::agency_client::AgencyClient;
+use aries_vcx_core::anoncreds::base_anoncreds::BaseAnonCreds;
+use aries_vcx_core::wallet::base_wallet::BaseWallet;
+use messages::msg_fields::protocols::cred_issuance::propose_credential::ProposeCredential;
+use messages::msg_fields::protocols::cred_issuance::CredentialIssuance;
+use messages::msg_fields::protocols::notification::Notification;
+use messages::AriesMessage;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+// todo: remove unused args
+pub async fn issuer_update_with_mediator(
+    sm: &mut Issuer,
+    _wallet: &Arc<dyn BaseWallet>,
+    _anoncreds: &Arc<dyn BaseAnonCreds>,
+    agency_client: &AgencyClient,
+    connection: &MediatedConnection,
+) -> VcxResult<IssuerState> {
+    trace!("Issuer::update_state >>>");
+    let messages = connection.get_messages(agency_client).await?;
+    if let Some((uid, msg)) = issuer_find_messages_to_handle(sm, messages) {
+        trace!("Issuer::update_state >>> found msg to handle; uid: {uid}, msg: {msg:?}");
+        sm.process_aries_msg(msg.into()).await?;
+        connection.update_message_status(&uid, agency_client).await?;
+    } else {
+        trace!("Issuer::update_state >>> found no msg to handle");
+    }
+    Ok(sm.get_state())
+}
+
+#[allow(clippy::unwrap_used)]
+pub fn issuer_find_messages_to_handle(
+    sm: &Issuer,
+    messages: HashMap<String, AriesMessage>,
+) -> Option<(String, AriesMessage)> {
```

Comment:
Are there any plans regarding removing stuff like this? IMO the protocol should have nothing to do with finding messages and even if we don't remove the mediated connection stuff, perhaps we can separate concerns a bit better?

Reply:
Definitely, it's only exists because:
-  `update_state` functions in `libvcx_core`, and these are also relied on in NodeJS tests
- integration tests for aries-vcx itself - as soon as IO removal is completed, and tests are not using mediated connection anymore (which will be possible with https://github.com/hyperledger/aries-vcx/pull/946 merged) we can get back to this, and at very least lift these `*_find_messages_to_handle` code-smells a level up to `libvcx_core`, where it can be (or not) dealt with separately.